core/src/main/scala/org/apache/spark/rdd/HadoopRDD.scala

```diff
@@ -220,8 +220,8 @@ class HadoopRDD[K, V](
 
       // Sets the thread local variable for the file's name
       split.inputSplit.value match {
-        case fs: FileSplit => SqlNewHadoopRDDState.setInputFileName(fs.getPath.toString)
-        case _ => SqlNewHadoopRDDState.unsetInputFileName()
+        case fs: FileSplit => FileScanRDDState.setInputFileName(fs.getPath.toString)
```

Comment:
@liancheng @yhuai Do you still use `HadoopRDD` to read data source relation? If not, I think we don't need to update file name here anymore.